### PR TITLE
FEATURE: app labels tagging

### PR DESCRIPTION
**Pull request by vikramdevtron**
# Description

this feature allow you to tag multiple tags to application, help you to categorize, and search your applications.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Backend Api's tested

- [x] Create App, with labels, without labels, empty labels etc.
- [x] Edit labels on apps, add and remove labels.
- [x] app meta info api.

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: api/restHandler/AppLabelsRestHandler.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	"github.com/devtron-labs/devtron/pkg/bean"
+	"github.com/devtron-labs/devtron/pkg/user"
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+	"gopkg.in/go-playground/validator.v9"
+	"net/http"
+	"strconv"
+)
+
+type AppLabelRestHandler interface {
```
Comment (nishant-d) [inline api/restHandler/AppLabelsRestHandler.go:32]: RBAC impl missing

File: docs/user-guide/app-labels.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+every app must have unique key.
+
+### 1. Add labels on creation of application
+
+Login with valid credentials and go to `Applications` and click on right top button `Add New App` and fill the required
+values.
+
+Labels are optionals and can be entered `key:values` format. multiple labels can be added without repeating `key` name.
+
+`TODO` : visual
```
Comment (nishant-d) [inline docs/user-guide/app-labels.md:19]: please add visuals and remove TODO

File: internal/sql/repository/pipelineConfig/AppLabelsRepository.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import (
+	"fmt"
+	"github.com/devtron-labs/devtron/internal/sql/models"
+	"github.com/go-pg/pg"
+)
+
+type AppLabel struct {
+	tableName struct{} `sql:"app_label" pg:",discard_unknown_columns"`
+	Id        int      `sql:"id,pk"`
+	AppId     int      `sql:"app_id"`
```
Comment (nishant-d) [inline internal/sql/repository/pipelineConfig/AppLabelsRepository.go:29]: can appId be null?

File: scripts/sql/21_create_app_label.up.sql [hunk 4]
```diff
@@ -0,0 +1,16 @@
+CREATE SEQUENCE IF NOT EXISTS id_seq_app_label;
+
+-- Table Definition
+CREATE TABLE "public"."app_label"
+(
+    "id"         int4 NOT NULL DEFAULT nextval('id_seq_app_label'::regclass),
```
Comment (nishant-d) [inline scripts/sql/21_create_app_label.up.sql:6]: not null constraint missing

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: api/restHandler/AppLabelsRestHandler.go (hunk 1 above)
Comment (vikramdevtron) [reply api/restHandler/AppLabelsRestHandler.go:33]: added

File: internal/sql/repository/pipelineConfig/AppLabelsRepository.go (hunk 3 above)
Comment (vikramdevtron) [reply internal/sql/repository/pipelineConfig/AppLabelsRepository.go]: added

File: scripts/sql/21_create_app_label.up.sql (hunk 4 above)
Comment (vikramdevtron) [reply scripts/sql/21_create_app_label.up.sql]: added

File: docs/user-guide/app-labels.md (hunk 2 above)
Comment (vikramdevtron) [reply docs/user-guide/app-labels.md]: added

**Review by nishant-d — APPROVED** [review]

**Pull request merged**